core_types/src/cas/storage.rs

```diff
@@ -327,47 +329,66 @@ impl EavTestSuite {
+            let eav = EntityAttributeValueIndex::new(&one.address(), &attribute, &many.address())
                 .expect("could not create EAV");
-            eav_storage.add_eav(&eav).expect("could not add eav");
-            expected.insert(eav);
+            eav_storage
+                .add_eav(&eav)
+                .expect("could not add eav")
+                .expect("could not add eav");
         }
 
         // throw an extra thing referencing many to show fetch ignores it
         let two = A::try_from_content(&foo_content)
             .expect("could not create AddressableContent from Content");
-        for many in vec![many_one.clone(), many_three.clone()] {
-            eav_storage
+        for many in vec![many_one.clone(), many_two.clone(), many_three.clone()] {
+            let eavi = eav_storage
                 .add_eav(
-                    &EntityAttributeValue::new(&two.address(), &attribute, &many.address())
+                    &EntityAttributeValueIndex::new(&two.address(), &attribute, &many.address())
                         .expect("Could not create eav"),
                 )
+                .expect("could not add eav")
                 .expect("could not add eav");
+            expected.insert(eavi);
         }
 
+        println!("expected {:?}", expected.clone());
+
         // show the many results for one
         assert_eq!(
             expected,
             eav_storage
-                .fetch_eav(Some(one.address()), Some(attribute.clone()), None)
-                .expect("could not fetch eav"),
+                .fetch_eav(
+                    Some(one.address()),
+                    Some(attribute.clone()),
+                    None,
+                    IndexQuery::default()
+                )
+                .expect("could not fetch eav")
         );
 
         // show one for the many results
         for many in vec![many_one.clone(), many_two.clone(), many_three.clone()] {
-            let mut expected_one = HashSet::new();
-            expected_one.insert(
-                EntityAttributeValue::new(&one.address(), &attribute.clone(), &many.address())
-                    .expect("Could not create eav"),
-            );
-            assert_eq!(
-                expected_one,
-                eav_storage
-                    .fetch_eav(None, Some(attribute.clone()), Some(many.address()))
-                    .expect("could not fetch eav"),
-            );
+            let mut expected_one = BTreeSet::new();
+            let eav =
+                EntityAttributeValueIndex::new(&one.address(), &attribute.clone(), &many.address())
+                    .expect("Could not create eav");
+            expected_one.insert(eav);
+            let fetch_set = eav_storage
+                .fetch_eav(
+                    None,
+                    Some(attribute.clone()),
+                    Some(many.address()),
+                    IndexQuery::default(),
```

Comment:
do we have tests for index queries other than `default`, e.g. range?

Reply:
I can definitely do add a ranged timestamp for sure, I was using the get_latest as a proof but a range as well would be good

Reply:
great

get_latest doesn't prove a range ;)